### PR TITLE
Fix regression - Add handling for NULL cells on encode_cell/1

### PR DESCRIPTION
**Pull request by alexmoore**

### CODE REVIEW

File: src/riak_pb_ts_codec.erl [hunk 1]
```diff
@@ -124,6 +124,8 @@ encode_cell(V) when is_boolean(V) ->
 encode_cell({time, V}) ->
     #tscell{timestamp_value = V};
 encode_cell(undefined) ->
+    #tscell{};
+encode_cell([]) -> %% NULL Cell
```
Comment (lukebakken) [inline src/riak_pb_ts_codec.erl]: @alexmoore what is the case where `[]` is the value? Is it on data insertion or retrieval?

Comment (alexmoore) [inline src/riak_pb_ts_codec.erl]: Retrieval. We put the `[]` in here: https://github.com/basho/riak_pb/pull/152/files#diff-5a85f13716f1297dfbd5fc46fd4c48ecR176

File: src/riak_pb_ts_codec.erl [hunk 2]
```diff
@@ -170,6 +172,5 @@ decode_cells([#tscell{binary_value    = undefined,
                       timestamp_value = undefined,
                       boolean_value   = undefined,
                       double_value    = undefined} | T], Acc) ->
-    %% all fields undefined: what are the circumstances where it is
-    %% possible/permitted?
+    %% NULL Cell
     decode_cells(T, [[] | Acc]).
```
Comment (lukebakken) [inline src/riak_pb_ts_codec.erl]: Whitespace at the beginning of the line.

Comment (alexmoore) [inline src/riak_pb_ts_codec.erl]: It's inline with the other recursive calls (see line [170](https://github.com/basho/riak_pb/pull/152#discussion-diff-43544377R170))

File: src/riak_pb_ts_codec.erl (hunk 1 above)
Comment (lukebakken) [inline src/riak_pb_ts_codec.erl]: OK, my concern is that special-casing `[]` is covering up a bug elsewhere. I feel like a list of cells should never have `[]` as one of its values.

File: src/riak_pb_ts_codec.erl (hunk 2 above)
Comment (lukebakken) [inline src/riak_pb_ts_codec.erl]: (Luke expands the view) ... Indeed it is.

File: src/riak_pb_ts_codec.erl (hunk 1 above)
Comment (lukebakken) [inline src/riak_pb_ts_codec.erl]: I see now ... the `[]` is used to represent a null `#tscell{}`.

Comment (alexmoore) [inline src/riak_pb_ts_codec.erl]: So the issue I found is within the generated ddl modules, they only accept `[]` for a null cell value. I think it used to be `undefined` but not sure if something happened.

Comment (lukebakken) [inline src/riak_pb_ts_codec.erl]: OK thank you. Can you please make a comment here and where that check is made, with perhaps a TODO to choose a better value?

**Comment by lukebakken** [comment]
+1 0a37595088ee6f35844bb4fd5a57693a5ca3a729

**Pull request merged**